addons/hr_recruitment/tests/test_recruitment_process.py

```diff
@@ -143,3 +142,73 @@ def test_email_application_multi_company(self):
         applicant = self.env['hr.applicant'].search([('email_from', 'ilike', 'Richard_Anderson@yahoo.com')], limit=1)
         self.assertEqual(applicant.company_id, other_company, 'Applicant should be created in the right company')
         self.assertEqual(applicant.candidate_id.company_id, other_company, 'Candidate should be created in the right company')
+
+    def test_multiple_emails_only_one_candidate(self):
+        """Make sure that receiving multiple emails from the same address does not create multiple candidates"""
+        job_developer = self.env["hr.job"].create(
+            {
+                "name": "Experienced Developer",
+            }
+        )
+        job_plumber = self.env["hr.job"].create(
+            {
+                "name": "Junior Plumber",
+            }
+        )
+
+        applicant_1_msg = """MIME-Version: 1.0
+Date: Thu, 19 Dec 2024 10:30:45 +0100
+Message-ID: <application1>
+Subject: Developer Application
+From:  Applicant 1 <applicant_1@example.com>
+To: hr@mycompany.com
+Content-Type: text/plain; charset="UTF-8"
+
+Hello, I want to be a developer.
+        """
+
+        applicant_2_msg = """MIME-Version: 1.0
+Date: Thu, 19 Dec 2024 15:30:00 +0100
+Message-ID: <application2>
+Subject: Plumber Application
+From:  Applicant 1 <applicant_1@example.com>
+To: hr@mycompany.com
+Content-Type: text/plain; charset="UTF-8"
+
+Hello, I want to be a plumber.
+        """
+
+        applicant_3_msg = """MIME-Version: 1.0
+Date: Thu, 19 Dec 2024 18:30:00 +0100
+Message-ID: <application3>
+Subject: Here is my application
+From:  Applicant 2 <applicant_2@example.com>
+To: hr@mycompany.com
+Content-Type: text/plain; charset="UTF-8"
+
+Hello, I want to work for you.
+        """
+
+        application_1_email = self.env["mail.thread"].message_process(
+            "hr.applicant", applicant_1_msg, custom_values={"job_id": job_developer.id}
+        )
+        application_2_email = self.env["mail.thread"].message_process(
+            "hr.applicant", applicant_2_msg, custom_values={"job_id": job_plumber.id}
+        )
+        application_3_email = self.env["mail.thread"].message_process(
+            "hr.applicant", applicant_3_msg, custom_values={"job_id": job_developer.id}
+        )
```

Comment:
For readability purpose, rename the variables so that we understand that those are application ids
```suggestion
        application_1_id = self.env["mail.thread"].message_process(
            "hr.applicant", applicant_1_msg, custom_values={"job_id": job_developer.id}
        )
        application_2_id = self.env["mail.thread"].message_process(
            "hr.applicant", applicant_2_msg, custom_values={"job_id": job_plumber.id}
        )
        application_3_id = self.env["mail.thread"].message_process(
            "hr.applicant", applicant_3_msg, custom_values={"job_id": job_developer.id}
        )
```